### PR TITLE
Add tiny tweaks for documentation, efficiency

### PR DESCRIPTION
**Pull request by kwalcock**
Close the source, run with assertions and then leave them commented for documentation, change vars to vals so they don't have to be double checked with each code inspection, simplify one takeWhile/lastOption.

### CODE REVIEW

**Comment by kwalcock** [comment]
Feel free to veto most of these.  52 document run without assertions being triggered, so for now I'm assuming it's correct.  Thanks for submitting to scrutiny of important code.

File: src/main/scala/org/clulab/wm/eidos/context/TimeNormFinder.scala [hunk 1]
```diff
@@ -48,27 +54,34 @@ class TimeNormFinder(parser: TemporalNeuralParser, timeRegexes: Seq[Regex]) exte
...
+          def nextOrElse(iterator: Iterator[Int], x: => Int): Int = if (iterator.hasNext) iterator.next else x
+          val start3 = nextOrElse(sentence.startOffsets.reverseIterator.map(_ - sentenceStart).dropWhile(_ > start2), 0)
+          val end3 = nextOrElse(sentence.endOffsets.iterator.map(_ - sentenceStart).dropWhile(_ < end2), sentenceText.length)
+          //assert(0 <= start3 && start3 <= start2)
+          //assert(end2 <= end3 && end3 <= sentenceText.length)
 
+          //assert(start3 <= end3)
           // yield the context interval
-          Interval(start, end)
+          Interval(start3, end3)
```
[the remark this answers is not in the document]
Comment (kwalcock) [reply src/main/scala/org/clulab/wm/eidos/context/TimeNormFinder.scala]: Definitely not critical.  The previous version is around to be asserted.  I periodically grep the code to see if vars were added unnecessarily (possibly blocking multithreading or imposing ordering constraints) or unintentionally (a common mistake, thanks JavaScript) and change them to vals.  Now I can ignore these.

[the remark this answers is not in the document]
Comment (kwalcock) [reply src/main/scala/org/clulab/wm/eidos/context/TimeNormFinder.scala]: Are you OK with this?  I was hoping you'd merge this and then I'd merge into master.

[the remark this answers is not in the document]
Comment (bethard) [reply src/main/scala/org/clulab/wm/eidos/context/TimeNormFinder.scala]: Yes, it's okay. I'll merge it tomorrow unless you beat me to it.

**Pull request merged**